### ISSUE TITLE
Update documentation to explain how variables are calculated

### ISSUE THREAD

**Issue by mdkf** [GitTools/GitVersion#984]
Please update the documentation to explain how the variables below are calculated:
CommitsSinceVersionSource
BuildMetaData
PreReleaseNumber

**Comment by JakeGinnivan** [comment]
The tldr is (and if you want to turn this into a PR that would be awesome):

 - CommitsSinceVersionSource: GitVersion uses the branch name, tags, GitVersion.yml and a few other things as version sources. It then counts the number of commits since then, often this can be thought of as number of commits since last tag, but that is not always the case
 - BuildMetaData: Includes things like branch, sha and other things. It will be appended to the version using `+<buildmetadata>` - See [http://semver.org/](http://semver.org/) for more info
 - PreReleaseNumber: It can be a few things depending on the mode, for continuous delivery, if you have a pre-release tag, the number will be incremented each time you tag your release/feature branch. For continuous delivery it will be incremented on each commit.

Hope that helps

**Comment by NickMoores** [comment]
> For continuous delivery it will be incremented on each commit.

@JakeGinnivan Should that be "For continuous deployment it will be incremented on each commit"?
